src/apps/survey/index.ts

```diff
@@ -14,26 +14,313 @@ import '../../plugins/cesium/ngv-plugin-cesium-slicing';
+      this.pointHighlightConfig = {
+        color: pointHighlightConf?.color
+          ? Color.fromCssColorString(pointHighlightConf?.color)
+          : Color.YELLOW,
+        outlineWidth:
+          typeof pointHighlightConf?.outlineWidth === 'number'
+            ? pointHighlightConf.outlineWidth
+            : 4,
+        outlineColor: pointHighlightConf?.outlineColor
+          ? Color.fromCssColorString(pointHighlightConf.outlineColor)
+          : Color.BLUE,
+        pixelSize: pointHighlightConf?.pixelSize || 7,
+        disableDepthTestDistance:
+          typeof pointHighlightConf?.disableDepthTestDistance === 'number'
+            ? pointHighlightConf?.disableDepthTestDistance
+            : undefined,
+      };
+      if (this.persistentDir) {
+        this.persistentDir = await getOrCreateDirectoryChain([
+          APP_NAME,
+          ...STORAGE_DIR,
+        ]);
+      }
+    },
+  });
+
+  initializeViewer(details: ViewerInitializedDetails): void {
+    this.viewer = details.viewer;
+    this.dataSourceCollection = details.dataSourceCollection;
+    this.dataSourceCollection
+      .add(this.dataSource)
+      .then(async () => {
+        if (!this.persistentDir) {
+          this.persistentDir = await getOrCreateDirectoryChain([
+            APP_NAME,
+            ...STORAGE_DIR,
+          ]);
+        }
+        this.surveys = await getJson<SurveysListItem[]>(
+          this.persistentDir,
+          STORAGE_LIST_NAME,
+        );
+        this.surveys.forEach((s) => {
+          const coords = {...s.coordinate};
+          const projection =
+            this.config.app.cesiumContext.clickInfoOptions?.projection;
+          if (projection) {
+            const projectedCoords = proj4(projection, 'EPSG:4326', [
+              coords.longitude,
+              coords.latitude,
+            ]);
+            coords.longitude = projectedCoords[0];
+            coords.latitude = projectedCoords[1];
+          }
+          const position = Cartesian3.fromDegrees(
+            coords.longitude,
+            coords.latitude,
+            coords.height,
+          );
+          this.addPoint(position, s.id);
+          this.viewer.scene.requestRender();
+        });
+      })
+      .catch((e) => console.error(e));
+    this.collections = details.primitiveCollections;
+    this.eventHandler = new ScreenSpaceEventHandler(this.viewer.canvas);
+    this.eventHandler.setInputAction(
+      (evt: ScreenSpaceEventHandler.PositionedEvent): void => {
+        const pickedObject: {id: Entity | undefined} = <
+          {id: Entity | undefined}
+        >this.viewer.scene.pick(evt.position);
+        if (
+          !pickedObject?.id?.id ||
+          !this.dataSource.entities.contains(pickedObject?.id)
+        ) {
+          return undefined;
+        }
+        this.editSurvey(pickedObject?.id?.id).catch((e) => console.error(e));
+      },
+      ScreenSpaceEventType.LEFT_CLICK,
```

Comment:
This should be moved to a dedicated function and only called when we change view (not at initialization).
There should be one list per view/place; so let's store in a subfolder using the view id.

Reply:
yes, that is what I am working on in https://camptocamp.atlassian.net/browse/GSHES-29